src/operators/ExtrudeStacked/QvisExtrudeStackedWindow.C

```diff
@@ -0,0 +1,857 @@
+    {
+        double val[3];
+        if(LineEditGetDoubles(axis, val, 3))
+            atts->SetAxis(val);
+        else
+        {
+            ResettingError(tr("Extrusion axis"),
+                DoublesToQString(atts->GetAxis(),3));
+            atts->SetAxis(atts->GetAxis());
+        }
+    }
+
+    // Do length
+    if(which_widget == ExtrudeStackedAttributes::ID_length || doAll)
+    {
+        double val;
+        if(LineEditGetDouble(length, val))
+            atts->SetLength(val);
+        else
+        {
+            ResettingError(tr("Length"),
+                DoubleToQString(atts->GetLength()));
+            atts->SetLength(atts->GetLength());
+        }
+    }
+
+    // Do steps
+    if(which_widget == ExtrudeStackedAttributes::ID_steps || doAll)
+    {
+        int val;
+        if(LineEditGetInt(steps, val))
+            atts->SetSteps(val);
+        else
+        {
+            ResettingError(tr("Number of steps"),
+                IntToQString(atts->GetSteps()));
+            atts->SetSteps(atts->GetSteps());
+        }
+    }
+}
+
+
+//
+// Qt Slot functions
+//
+
+
+void
+QvisExtrudeStackedWindow::axisProcessText()
+{
+    GetCurrentValues(ExtrudeStackedAttributes::ID_axis);
+    Apply();
+}
+
+void
+QvisExtrudeStackedWindow::lengthProcessText()
+{
+    GetCurrentValues(ExtrudeStackedAttributes::ID_length);
+    Apply();
+}
+
+
+void
+QvisExtrudeStackedWindow::stepsProcessText()
+{
+    GetCurrentValues(ExtrudeStackedAttributes::ID_steps);
+    Apply();
+}
+
+
+void
+QvisExtrudeStackedWindow::preserveOriginalCellNumbersChanged(bool val)
+{
+    atts->SetPreserveOriginalCellNumbers(val);
+    SetUpdate(false);
+    Apply();
+}
+
+// ****************************************************************************
+//  Method:  QvisExtrudeStackedWindow::variableGroupChanged
+//
+//  Purpose:
+//    extrude by variable(s)
+//
+//  Arguments:
+//    val   boolean true - extrude by variable
+//
+//  Programmer:  Jeremy Meredith
+//  Creation:    March 16, 2007
+//
+// ****************************************************************************
+
+void
+QvisExtrudeStackedWindow::variableGroupChanged(bool val)
+{
+    atts->SetByVariable(val);
+
+    if(atts->GetByVariable())
+      lengthLabel->setText("Scale Factor");
+    else
+      lengthLabel->setText("Length");
+
+    variableSelected(nullptr);
+
+    SetUpdate(false);
+    Apply();
+}
+
+// ****************************************************************************
+//  Method:  QvisExtrudeStackedWindow::addVariable
+//
+//  Purpose:
+//    add a variable (i.e. variable) to the plot
+//
+//  Arguments:
+//    variableToAdd   the name of the new variable to add
+//
+//  Programmer:  Jeremy Meredith
+//  Creation:    March 16, 2007
+//
+// ****************************************************************************
+
+void
+QvisExtrudeStackedWindow::addVariable(const QString &variableToAdd)
+{
+    atts->InsertVariable(variableToAdd.toStdString());
+    //SetUpdate(false);
```

Comment:
If the SetUpdate call isn't needed, please remove it.

Reply:
Done